### PR TITLE
[cluster-test] Clone vault image to ECR and pull from ECR instead of …

### PR DESCRIPTION
**Pull request by snakech**
…dockerhub

related pr: novifinancial/libra-ops#901 for create new vault repo on ECR

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/master/developers.diem.com, and link to your PR here.)

### CODE REVIEW

**Comment by snakech** [comment]
please let me know if I missed any corresponded places need to change. Also may I ask if there has any test plan? Do I need to merge pr to create new repo first and then do CT test with this PR?

File: docker/dockerhub_to_novi_ecr.sh [hunk 1]
```diff
@@ -57,6 +57,7 @@ docker pull --disable-content-trust=false docker.io/libra/validator:"$DOCKERHUB_
 docker pull --disable-content-trust=false docker.io/libra/validator_tcb:"$DOCKERHUB_TAG"
 docker pull --disable-content-trust=false docker.io/libra/cluster_test:"$DOCKERHUB_TAG"
 docker pull --disable-content-trust=false docker.io/libra/client:"$DOCKERHUB_TAG"
+docker pull --disable-content-trust=false docker.io/libra/vault:"1.4.0"
```
Comment (sausagee) [inline docker/dockerhub_to_novi_ecr.sh:60]: This should be just `vault`.

File: docker/dockerhub_to_novi_ecr.sh [hunk 2]
```diff
@@ -76,5 +77,6 @@ docker push --disable-content-trust=true "$TARGET_REPO"/diem/validator:"$OUTPUT_
 docker push --disable-content-trust=true "$TARGET_REPO"/diem/validator_tcb:"$OUTPUT_TAG"
 docker push --disable-content-trust=true "$TARGET_REPO"/diem/cluster_test:"$OUTPUT_TAG"
 docker push --disable-content-trust=true "$TARGET_REPO"/diem/client:"$OUTPUT_TAG"
+docker push --disable-content-trust=true "$TARGET_REPO"/diem/vault:"1.4.0"
```
Comment (sausagee) [inline docker/dockerhub_to_novi_ecr.sh:80]: same here and the two references below.

**Review by sausagee — CHANGES_REQUESTED** [review]

**Review by sausagee — APPROVED** [review]

File: docker/dockerhub_to_novi_ecr.sh [hunk 3]
```diff
@@ -57,6 +57,7 @@ docker pull --disable-content-trust=false docker.io/libra/validator:"$DOCKERHUB_
 docker pull --disable-content-trust=false docker.io/libra/validator_tcb:"$DOCKERHUB_TAG"
 docker pull --disable-content-trust=false docker.io/libra/cluster_test:"$DOCKERHUB_TAG"
 docker pull --disable-content-trust=false docker.io/libra/client:"$DOCKERHUB_TAG"
+docker pull --disable-content-trust=false docker.io/vault:"1.4.0"
```
Comment (rexhoffman) [inline docker/dockerhub_to_novi_ecr.sh:60]: Given this file has no hard coded version numbers I'm loath to have it take them on now.   At the very least this should be input for the file, at best there should be a single file that acts as the source of truth for the vault version, and given the sensitivity of the software perhaps a sha version pull as well, like in all our debian base versions in our docker builds.    But by having that single file perhaps RUNTIME_SERVICES, set up with things like vault=1.4.0@sha256:<sha> we could use it as input here to make sure we pull the exact file we want, and elsewhere use that version number as a way to reference it.

**Comment by rexhoffman** [comment]
I wont block this, not knowing the severity of the issue, or your workload.   But.   This is adding cut&paste version numbers to a release file that never had them before.   This adds debt that will eventually cause broken releases, confusion and pain unless someone else pays down that debt for you.    I left a comment on how I think you can minimize that debt if you must add it, but I must admit I was very much hoping you'd figure out the k8s pull so that CI would have to move these files which it is not signing, or tagging.    This is pure debt being added due to existing debt in k8s.

**Comment by snakech** [comment]
> I wont block this, not knowing the severity of the issue, or your workload. But. This is adding cut&paste version numbers to a release file that never had them before. This adds debt that will eventually cause broken releases, confusion and pain unless someone else pays down that debt for you. I left a comment on how I think you can minimize that debt if you must add it, but I must admit I was very much hoping you'd figure out the k8s pull so that CI would not have to move these files which it is not signing, or tagging. This is pure debt being added due to existing debt in k8s.

Thx for reviewing and suggestion, this is a prety temp solution for now to unblock the LBT and make pr landing works again. Login to k8s and remove all hard code version is the next step I will do

**Comment by snakech** [comment]
Since we do rarely update vault image, I manually tag an image vault-1.4.0 and push to ecr instead of clone image every time for each LBT run

**Comment by snakech** [comment]
tested with 
```
./scripts/cti -T dev_czh_pull_7328 --suite land_blocking
2021-01-25T23:58:50.092234Z [main] INFO testsuite/cluster-test/src/aws.rs:28 Scaling to desired_capacity : 0, buffer: 0, asg_name: ct-0-k8s-testnet-validators
all up : 924 TPS, 4898 ms latency, 6450 ms p99 latency, no expired txns
```

**Pull request merged**